ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -799,6 +811,27 @@ private boolean isInnerNode(final PlanNode node) {
     throw new IllegalStateException("Unknown node type: " + node.getClass().getName());
   }
 
+  private JoinKey buildForeignJoinKey(final Join join, final ColumnReferenceExp columnRef) {
+    final AliasedDataSource leftSource = join.getInfo().getLeftSource();
+    final SourceName alias = leftSource.getAlias();
+    final List<QualifiedColumnReferenceExp> leftSourceKeys =
+        leftSource.getDataSource().getSchema().key().stream()
+            .map(c -> new QualifiedColumnReferenceExp(alias, c.name()))
+            .collect(Collectors.toList());
+
+    final QualifiedColumnReferenceExp keyColumn = leftSourceKeys.get(0);
+
+    final ColumnName keyColumnName = ColumnNames
+        .generatedJoinColumnAlias(keyColumn.getQualifier(), keyColumn.getColumnName());
```

Comment:
See my comment below about the purpose of `keyColumn` in `ForeignJoinKey`. I think this code can be removed.